source/jormungandr/jormungandr/street_network/handimap.py

```diff
@@ -0,0 +1,336 @@
+
+        self.check_content_response(resp_json, origins, destinations)
+        return self._create_matrix_response(resp_json, origins, destinations, max_duration)
+
+    def _direct_path(
+        self,
+        instance,
+        mode,
+        pt_object_origin,
+        pt_object_destination,
+        fallback_extremity,
+        request,
+        direct_path_type,
+        request_id,
+    ):
+        if mode != "walking":
+            self.log.error('Handimap, mode {} not implemented'.format(mode))
+            raise InvalidArguments('Handimap, mode {} not implemented'.format(mode))
+
+        walking_speed = request["walking_speed"]
+        language = self.get_language_parameter(request)
+
+        params = self._make_request_arguments_direct_path(
+            pt_object_origin, pt_object_destination, walking_speed, language
+        )
+
+        response = self._call_handimap('route', params)
+        self.check_response(response)
+        return self._get_response(
+            ujson.loads(response.text), pt_object_origin, pt_object_destination, fallback_extremity
+        )
+
+    @classmethod
+    def _get_response(cls, json_response, pt_object_origin, pt_object_destination, fallback_extremity):
+        '''
+        :param fallback_extremity: is a PeriodExtremity (a datetime and it's meaning on the fallback period)
+        '''
+        resp = response_pb2.Response()
+        resp.status_code = 200
+        resp.response_type = response_pb2.ITINERARY_FOUND
+        handimap_trip = json_response["trip"]
+        journey = resp.journeys.add()
+        journey.duration = int(round(handimap_trip['summary']["time"]))
+        datetime, represents_start_fallback = fallback_extremity
+        if represents_start_fallback:
+            journey.departure_date_time = datetime
+            journey.arrival_date_time = datetime + journey.duration
+        else:
+            journey.departure_date_time = datetime - journey.duration
+            journey.arrival_date_time = datetime
+        journey.durations.total = journey.duration
+        journey.durations.walking = journey.duration
+
+        journey.distances.walking = int(round(kilometers_to_meters(handimap_trip['summary']["length"])))
+
+        previous_section_endtime = journey.departure_date_time
+        for index, handimap_leg in enumerate(handimap_trip['legs']):
+            section = journey.sections.add()
+            section.type = response_pb2.STREET_NETWORK
+            section.duration = int(round(handimap_leg["summary"]['time']))
+            section.begin_date_time = previous_section_endtime
+            section.end_date_time = section.begin_date_time + section.duration
+            previous_section_endtime = section.end_date_time
+
+            section.id = 'section_{}'.format(index)
+            section.length = int(round(kilometers_to_meters(handimap_leg["summary"]['length'])))
+
+            if index == 0:
```

Comment:
to avoid branches (`if`), you can do operations to the specific items at the end..